File: app/src/main/java/org/lifefortheorc/tudounotepad/widget/TudouToast.java
package org.lifefortheorc.tudounotepad.widget;

import android.content.Context;
import android.widget.Toast;

/**
 * Created by geminiwen on 15/7/2.
 */
public class TudouToast {
    private static Context sAppContext;
    private static Toast sToast;

    public static void init(Context c) {
        sAppContext = c;
    }


    public static void show(String text, int duration) {
        if (sToast != null) {
            sToast.cancel();
        }
        sToast = Toast.makeText(sAppContext, text, duration);
        sToast.show();
    }

    public static void show(String text) {
        TudouToast.show(text, Toast.LENGTH_LONG);
    }

    public static void show(int text, int duration) {
        if (sToast != null) {
            sToast.cancel();
        }
        sToast = Toast.makeText(sAppContext, text, duration);
        sToast.show();
    }

    public static void show(int text) {
        TudouToast.show(text, Toast.LENGTH_LONG);
    }
}

File: app/src/main/kotlin/org/lifefortheorc/tudounotepad/widget/TudouToast.kt
package org.lifefortheorc.tudounotepad.widget

import android.content.Context
import android.widget.Toast
import org.lifefortheorc.tudounotepad.widget.TudouToast

/**
 * Created by geminiwen on 15/7/13.
 */

object TudouToast {
    var context: Context? = null
    var toast: Toast? = null

    public fun init(ctx : Context) {
        this.context = ctx;
    }

    public fun show(text: String, duration: Int) {
        toast?.cancel()
        toast = Toast.makeText(context, text, duration)
        toast?.show()
    }

    public fun show(text: String) {
        show(text, Toast.LENGTH_LONG)
    }

    public fun show(text: Int, duration: Int) {
        toast?.cancel()
        toast = Toast.makeText(context, text, duration)
        toast?.show()
    }

    public fun show(text: Int) {
        show(text, Toast.LENGTH_LONG)
    }
}
